go/pkg/tool/tool.go

```diff
@@ -121,35 +136,32 @@ func (c *Client) prepCommand(ctx context.Context, client *rexec.Client, actionDi
 		inputPaths = append(inputPaths, f.Name())
 	}
 	// Construct Command object.
-	cmd := commandFromREProto(commandProto)
+	cmd := command.FromREProto(commandProto)
 	cmd.InputSpec.Inputs = inputPaths
+	cmd.InputSpec.InputNodeProperties = nodeProperties
 	cmd.ExecRoot = inputRoot
 	if actionProto.Timeout != nil {
 		cmd.Timeout = actionProto.Timeout.AsDuration()
 	}
 	return cmd, nil
 }
 
-func commandFromREProto(cmdPb *repb.Command) *command.Command {
-	cmd := &command.Command{
-		InputSpec: &command.InputSpec{
-			EnvironmentVariables: make(map[string]string),
-		},
-		Identifiers: &command.Identifiers{},
-		WorkingDir:  cmdPb.WorkingDirectory,
-		OutputFiles: cmdPb.OutputFiles,
-		OutputDirs:  cmdPb.OutputDirectories,
-		Platform:    make(map[string]string),
-		Args:        cmdPb.Arguments,
-	}
-
-	for _, ev := range cmdPb.EnvironmentVariables {
-		cmd.InputSpec.EnvironmentVariables[ev.Name] = ev.Value
+func readNodePropertiesFromFile(path string) (nps map[string]*repb.NodeProperties, err error) {
+	if _, err = os.Stat(path); err == nil {
```

Comment:
Can you rearrange the conditions such that [the terminal one appears first](https://google.github.io/styleguide/go/decisions#indent-error-flow:~:text=terminal%20condition%20is%20not%20met%20should%20appear%20after%20the%20if%20block)?